### DISCUSSION TITLE
Viewing model weights

### DISCUSSION THREAD

**Discussion by andrew11221** [microsoft/onnxruntime#8391] [Other Q&A] [answered]
Is there a way to print certain weights from an onnx model through the ORT C/C++ API?

**Comment by pranavsharma** [comment, marked as answer]
There's no C/C++ API to do so. The onnx model is simply a protobuf file. Just read it using any protobuf language binding and dump the weights.

**Reply by andrew11221** [reply]
I see, thank you. Unfortunately, my goal was to verify that ORT itself loads the model properly since I've made some modifications to the source code. Maybe I can find a different way.

**Reply by andrew11221** [reply]
~Is there an easy way to print out the output at specific layers in the C/C++ API?~ https://github.com/microsoft/onnxruntime/issues/1455

**Reply by pranavsharma** [reply]
You can always look at the output of the individual nodes or the entire graph itself before and after to verify. Possibly this might help https://www.onnxruntime.ai/docs/how-to/build/inferencing.html#debugnodeinputsoutputs.

**Reply by andrew11221** [reply]
Thanks Pranav!
